org.lflang/src/org/lflang/generator/GeneratorBase.xtend

```diff
@@ -1522,6 +1551,12 @@ abstract class GeneratorBase extends AbstractLFValidator {
                         /* FIXME: The at keyword should support a directory component.
                          * federateInstance.dir = instantiation.host.dir
                          */
+                        if (federateInstance.host !== null && 
```

Comment:
I think there may be more cases here. What about 127.0.0.1 or really any IP that a local device could have been assigned. It may be difficult to come up with a robust platform-independent solution for this, however...

Reply:
I think we intentionally left 127.0.0.1 out with the hopes that we could test remote federated features. It shouldn't materially make any difference, except that clock sync might get enabled for federates marked as remote.